packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -307,56 +346,69 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
         const dataModel = await LoadTestDataStoreModel.createRunnerInstance(
             config, reset, this.root, this.runtime, this.context.containerRuntime);
 
-         // At every moment, we want half the client to be concurrent writers, and start and stop
-        // in a rotation fashion for every cycle.
-        // To set that up we start each client in a staggered way, each will independently go thru write
-        // and listen cycles
-
-        const cycleMs = config.testConfig.readWriteCycleMs;
-        let t: NodeJS.Timeout | undefined;
-        if(config.verbose) {
+        let timeout: NodeJS.Timeout | undefined;
+        if (config.verbose) {
             const printProgress = () => {
                 dataModel.printStatus();
-                t = setTimeout(printProgress, config.testConfig.progressIntervalMs);
+                timeout = setTimeout(printProgress, config.testConfig.progressIntervalMs);
             };
-            t = setTimeout(printProgress, config.testConfig.progressIntervalMs);
+            timeout = setTimeout(printProgress, config.testConfig.progressIntervalMs);
         }
 
+        // At every moment, we want half the clients to be concurrent writers, and switch every cycle.
+        // To set that up we pair each client with another, and half of the clients will wait for their partner to
+        // finish before writing.
+        const cycleMs = config.testConfig.readWriteCycleMs;
         const clientSendCount = config.testConfig.totalSendCount / config.testConfig.numClients;
         const opsPerCycle = config.testConfig.opRatePerMin * cycleMs / 60000;
         const opsGapMs = cycleMs / opsPerCycle;
-        try{
+        try {
+            let blobP: Promise<void> | undefined;
             while (dataModel.counter.value < clientSendCount && !this.disposed) {
                 // this enables a quick ramp down. due to restart, some clients can lag
-                // leading to a slow ramp down. so if there are less than half the clients
-                // and it's partner is done, return true to complete the runner.
-                if(this.runtime.getAudience().getMembers().size < config.testConfig.numClients / 2
+                // leading to a slow ramp down. so if there are fewer than half the clients
+                // and our partner is done, return true to complete the runner.
+                if (this.runtime.getAudience().getMembers().size < config.testConfig.numClients / 2
                     && ((await dataModel.getPartnerCounter())?.value ?? 0) >= clientSendCount) {
                     return true;
                 }
 
-                if(dataModel.haveTaskLock()) {
+                if (dataModel.haveTaskLock()) {
                     dataModel.counter.increment(1);
+
+                    // write a blob once per cycle
+                    if (dataModel.isBlobWriter && !blobP) {
+                        blobP = dataModel.writeBlob();
```

Comment:
this is going to end up writing a lot of blobs when we run for a long time. I think we need a config that more easily controls the total number of blobs written